### PR TITLE
Add snippet/highlight/pagination integration tests

### PR DESCRIPTION
**Pull request by bidoubiwa**
see #389 

Unit tests have also been added see: 
#387

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -35,7 +35,11 @@ export type IMResponse = {
   nbPages?: number
 }
 
-export type SearchResponse = IStypes.SearchResponse & IMResponse
+export type SearchResponse = Omit<IStypes.SearchResponse, 'hits'> &
```
Comment (curquiza) [inline src/types/types.ts:38]: Just for my knowledge, what does the `&` mean? (sorry not to have asked before, I'm sure it's obvious for you)

**Review by curquiza — APPROVED** [review]
Thanks a lot @bidoubiwa for these tests, you save us!

File: src/types/types.ts (hunk 1 above)
Comment (bidoubiwa) [reply src/types/types.ts:38]: It is an intersection between the types ([see doc for great examples](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces))

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**